File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.kt
class Solution {
    fun solution(n: Int, left: Long, right: Long): IntArray {
        val size = (right - left + 1).toInt()
        val answer = IntArray(size)

        for (i in 0 until size) {
            val idx = left + i
            val row = idx / n
            val col = idx % n
            answer[i] = (if (row > col) row else col).toInt() + 1
        }

        return answer
    }
}

File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int n, long long left, long long right) {
    int size = right - left + 1;
    int* answer = (int*)malloc(sizeof(int) * size);
    
    for (long long i = left; i <= right; i++) {
        long long row = i / n;
        long long col = i % n;
        answer[i - left] = (row > col ? row : col) + 1;
    }
    
    return answer;
}
